### ISSUE TITLE
Pro Version - How to pass custom arguments?

### ISSUE THREAD

**Issue by jamesautodude** [wire-elements/modal#471]
I'm sure I'm just overlooking this, but I looked through and tried everything...

I'm trying to do something like this:

`wire:click="$dispatch('modal.open', {component: 'Role2.create-event', arguments: {'selectedCalendarDate': {{ $currentDayForModal }}} })"`

But I'm not receiving the correct data on the component under `$this->selectedCalendarDate`

Do I need to do @json or something?

Thank you in advance!

**Comment by PhiloNL** [comment]
Hi @jamesautodude,

Tried this in the demo repository and it works fine:

```blade
<button type="button" wire:modal="assets-show, @js(['releaseId' => $release->id, 'lazy' => true, 'hello' => 'bar'])">View Assets</button>
```

```php
class AssetsShow extends Modal
{
    public int $releaseId;

    public $hello;

    public function mount()
    {
        dd($this->hello);
    }
}
```

**Comment by jamesautodude** [comment]
> Hi @jamesautodude,
> 
> Tried this in the demo repository and it works fine:
> 
> ```blade
> <button type="button" wire:modal="assets-show, @js(['releaseId' => $release->id, 'lazy' => true, 'hello' => 'bar'])">View Assets</button>
> ```
> 
> ```php
> class AssetsShow extends Modal
> {
>     public int $releaseId;
> 
>     public $hello;
> 
>     public function mount()
>     {
>         dd($this->hello);
>     }
> }
> ```

Thank you so much 🙏

**Comment by jamesautodude** [comment]
It looks like this has a problem rehydrating the @js when parent modal is re-rendered

My example:

- Parent modal has a foreach loop with users
- I add users to the foreach loop for the parent modal
- Each user has a delete button within the foreach loop, to delete the user from the parent modal.
- As long as I don't add new users, the delete buttons work as expected.
- if I add a user and re-render the component (& parent modal), the delete button's user ID (within @js) doesn't match the new;y created user.
- Once I hard refresh the whole page, the IDs within @js match again.

Basically @js doesn't recognize newly created users (or re-renders).

Any ideas?

**Comment by jamesautodude** [comment]
Disregard, fixed it with using:

`wire:click="$dispatch('modal.open', {component: 'Role2.Calendar.remove-employee', arguments: {'eventId': {{ $eventId }}, 'userId': {{ $user->id }} }})"`

I can't believe I keep overlooking the documentation :/

**Comment by theticraft** [comment]
Is `wire:modal` exclusive to the pro version?
